### ISSUE TITLE
Breaking changes - IdwsFileSystem

### ISSUE THREAD

**Issue by andersmelander** [EricGrange/DWScript#8]
Commit f0e53d5baf98d8d34cb4418d529ac26003733cc3 contains a breaking change to the `IdwsFileSystem` interface:

https://bitbucket.org/egrange/dwscript/commits/f0e53d5baf98d8d34cb4418d529ac26003733cc3#LSource/dwsFileSystem.pasF51T54

This is pretty annoying when one already have custom file systems implemented as the change breaks existing code without any benefit.
While the added interface methods might be convenient in new code, for existing code they are just overhead as one will have already implemented the same functionality in some other way if it was needed. I already have empty methods implementation stubs after the last breaking change to this interface.

Since it’s an interface would it not be possible to extend it with separate feature interfaces instead?

```Delphi
type
  IdwsFileSystemDirectoryExists = interface
  [{blahblah}]
    function DirectoryExists(const directoryName : TFilename) : Boolean;
  end;
  [...]
begin
  var FileSystemDirectoryExists: IdwsFileSystemDirectoryExists;
  if (Supports(MyFileSystem, IdwsFileSystemDirectoryExists, FileSystemDirectoryExists)) then
  begin
    if (FileSystemDirectoryExists.DirectoryExists(directoryName)) then
      [...]
  end;
end;
```
That’s what I already do with my different file system implementations when I need a feature that isn’t provided by the base file system API. It saves me from having to implement empty stubs in the file systems where a given feature isn’t relevant.

**Comment by EricGrange** [comment]
Sorry, I did not expect for that interface to be implemented externally, I'll split the new functionality.

It's not committed yet, but all direct calls from built-in script libraries to file system will go through it eventually: there will be no more direct use of SysUtils, IOUtils etc. file system functions from built-in script functions. The goal being to facilitate full containerization of scripts (either for use in portable apps or to restrict to a db, zip or whatever)

**Comment by EricGrange** [comment]
Fixed by 58b12d3c89cacc6549d3baac9d9412e9897520e7
